src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java

```diff
@@ -50,12 +50,27 @@
  * A token character can be repeated to ensure that the field occupies a certain minimum
  * size. Values will be left-padded with 0 unless padding is disabled in the method invocation.
  * <br>
- * Tokens can be marked as optional by surrounding them with brackets [ ]. These tokens will 
- * only be printed if the token value is non-zero. Any literals within optional blocks will only be 
- * printed if the nearest prior non-literal token value was non-zero.
+ * Tokens can be marked as optional by surrounding them with brackets [ ]. These tokens will
+ * only be printed if the token value is non-zero. Literals within optional blocks will only be
+ * printed if the preceding non-literal token is non-zero. Leading optional literals will only
+ * be printed if the following non-literal is non-zero.
+ * Multiple optional blocks can be used to group literals with the desired token. Note that
+ * multiple optional tokens without literals can result in impossible to understand output.
+ * (See examples below)
  * <br>
+ * <table border="1">
+ * <caption>Example Output</caption>
+ * <tr><th>pattern</th><th>Duration.ofDays(1)</th><th>Duration.ofHours(1)</th><th>Duration.ofMinutes(1)</th></tr>
```

Comment:
This is nice. For the entry with all optional formatting `[d'd'H'h'm'm's's']` it would be useful to state what happens when the duration is 0. E.g. If the entire format is optional and the duration is zero the result is the empty string.

Reply:
Agree. Added a column for Duration.ofMillis(0) so that empty string results are noticeable, as well as a specific Note.

Reply:
> Agree. Added a column for Duration.ofMillis(0) so that empty string results are noticeable, as well as a specific Note.

Is the 0 important or the Millis part? It might be more obvious to refer to Duration.ZERO.

Reply:
You are correct. I hadn't checked for that static field. Will update.